src/histolab/filters/image_filters.py

```diff
@@ -0,0 +1,700 @@
+from PIL import ImageOps
+
+from src.histolab.filters import image_filters_functional as F
+
+
+class Compose(object):
+    """Composes several filters together.
+
+    Parameters
+    ----------
+    filters : list of Filters
+        List of filters to compose
+    """
+
+    def __init__(self, filters):
+        self.filters = filters
+
+    def __call__(self, img):
+        for f in self.filters:
+            img = f(img)
+        return img
+
+
+class Invert(object):
+    """Invert an image."""
+
+    def __call__(self, img):
+        """
+        Parameters
+        ----------
+        img : PIL.Image.Image
+            Input image
+
+        Returns
+        -------
+        PIL.Image.Image
+            Inverted image
+        """
+        return F.invert(img)
+
+    def __repr__(self):
+        return self.__class__.__name__ + "()"
+
+
+class RgbToGrayscale(object):
+    """Convert an RGB image to a grayscale image."""
+
+    def __call__(self, img):
+        """Convert an RGB image to a grayscale image.
+
+        Parameters
+        ----------
+        img : PIL.Image.Image
+            Input image
+
+        Returns
+        -------
+        PIL.Image.Image
+            Grayscale image
+        """
+        return ImageOps.grayscale(img)
+
+    def __repr__(self):
+        return self.__class__.__name__ + "()"
+
+
+class RgbToHed(object):
+    """Convert RGB channels to HED channels."""
+
+    def __call__(self, img):
+        """Convert RGB channels to
+        HED (Hematoxylin - Eosin - Diaminobenzidine) channels.
+
+        Parameters
+        ----------
+        img : PIL.Image.Image
+            Input image
+
+        Returns
+        -------
+        PIL.Image.Image
+            Image in HED space
+        """
+        hed = F.rgb_to_hed(img)
+        return hed
+
+    def __repr__(self):
+        return self.__class__.__name__ + "()"
+
+
+class RgbToHsv(object):
+    """Convert RGB channels to HSV channels"""
+
+    def __call__(self, img):
+        """Convert RGB channels to
+        HED (Hematoxylin - Eosin - Diaminobenzidine) channels.
+
+        Parameters
+        ----------
+        img : PIL.Image.Image
+            Input image
+
+        Returns
+        -------
+        PIL.Image.Image
+            Image in HED space
+        """
+        hsv = F.rgb_to_hsv(img)
+        return hsv
+
+    def __repr__(self):
+        return self.__class__.__name__ + "()"
+
+
+class StretchContrast(object):
+    """Increase contrast in image based on intensities in
+    a specified range"""
```

Comment:
the first line of the docstring should be 1 line with a summary of the behaviour:
e.g.
```
""" Increase contrast in image based on intensities in a specified range"""
```
If it exceeds the 88 chars you have to summarize that and add a detailed information in another line

```
"""description bla bla bla

details
"""